src/App.tsx

```diff
@@ -1,24 +1,212 @@
 
-const USER_ID = 0;
+const USER_ID = 12041;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<Errors | null>(null);
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [selectedTodos, setSelectedTodos] = useState<number[]>([0]);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const titleField = useRef<HTMLInputElement>(null);
+  const form = useRef<HTMLFormElement>(null);
+  // #region "useEffects"
+
+  const handleError = (err: Errors) => {
+    setError(err);
+
+    setTimeout(() => {
+      setError(null);
+    }, 3000);
+  };
+
+  const handleDelete = async (todoId: number) => {
+    try {
+      if (error) {
+        setError(null);
+      }
+
+      setSelectedTodos(prev => [...prev, todoId]);
+
+      await deleteTodo(todoId);
+
+      setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+    } catch (e) {
+      setSelectedTodos(prev => [...prev, 0]);
+      handleError(Errors.UnableToDelete);
+    }
+  };
+
+  const onErrorNotifyClose = () => {
+    setError(null);
+  };
+
+  const onFilterSelect = (filterT: FilterType) => {
+    setFilterType(filterT);
+  };
+
+  useEffect(() => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+  });
+
+  useEffect(() => {
+    const renderTodos = async () => {
+      try {
+        const todosFromServer = await getTodos(USER_ID);
+
+        setTodos(todosFromServer);
+      } catch (e) {
+        handleError(Errors.UnableToLoad);
+
+        throw e;
+      }
+    };
+
+    renderTodos();
+  }, []);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const title = event.currentTarget.titleInput.value.trim();
+
+    if (!title) {
+      handleError(Errors.EmptyTitle);
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      title,
+      userId: USER_ID,
+      completed: false,
+    });
+
+    const postingTodo = async () => {
+      try {
+        if (error) {
+          setError(null);
+        }
+
+        setIsSubmitting(true);
+
+        const response = await postTodo({
+          title,
+          userId: USER_ID,
+          completed: false,
+        });
+
+        setTempTodo(null);
+
+        setTodos(prevTodos => [...prevTodos, response]);
+        form.current?.reset();
+      } catch (err) {
+        handleError(Errors.UnableToAdd);
+
+        setTempTodo(null);
+        throw err;
+      } finally {
+        setIsSubmitting(false);
+      }
+    };
+
+    postingTodo();
+  };
+
+  const clearCompleted = useCallback(() => {
+    const completedTodos = todos
+      .filter((todo) => todo.completed);
+
+    setSelectedTodos((currentIds) => [
+      ...currentIds,
+      ...completedTodos.map((completedTodo) => completedTodo.id),
+    ]);
```

Comment:
After you get a response from API you should remove ids of todos that were deleted from your loading state

Reply:
> After you get a response from API you should remove ids of todos that were deleted from your loading state

Not fixed